src/cosalib/qemuvariants.py

```diff
@@ -304,24 +304,32 @@ def mutate_image(self):
             log.info(f"Moving {work_img} to {final_img}")
             shutil.move(work_img, final_img)
 
-        if self.gzip:
+        if self.compression == "skip":
+            meta_patch.update({
+                'skip-compression': True
+            })
+        elif self.compression is not None:
             sha256 = sha256sum_file(final_img)
             size = os.stat(final_img).st_size
             temp_path = f"{final_img}.tmp"
+
+            match self.compression:
+                case "gzip":
+                    rc = ['gzip', '-9c', final_img]
+                case "zip":
+                    rc = ['zip', '-9', "-", final_img ]
+                    # `zip` does not append ".zip". Doing so manually for now
+                    final_img += ".zip"
```

Comment:
Please stop hacking around the incorrect `image_suffix` and actually change `image_suffix` as [requested](https://github.com/coreos/coreos-assembler/pull/3363#discussion_r1111614163).